homeassistant/components/tomorrowio/sensor.py

```diff
@@ -289,6 +292,32 @@ async def async_setup_entry(
     async_add_entities(entities)
 
 
+def smart_round(value: float) -> float:
+    """Round a float to have at least two digits."""
+    if float(value) == 0.0:
+        return float(value)
+
+    num_digits = 2
+    if (new_val := round(value, num_digits)) != 0.0:
+        return new_val
+
+    while num_digits < 5 and (new_val := round(value, num_digits)) == 0.0:
+        num_digits += 1
```

Comment:
Calculate the logarithm with base 10 of the value, then use that to round, instead of this loop.